packages/amd-loader/amd-loader.js

```diff
@@ -0,0 +1,196 @@
+'use strict';
+
+(function() {
+
+function Registry() {
+  this.modules = {};
```

Comment:
Use `Object.create(null)` for a map-style object, otherwise stuff like `this.modules['valueOf']` will be truthy.